src/App.tsx

```diff
@@ -1,26 +1,193 @@
+        setLoading(false);
+        setLoadingTodoId(null);
+        if (inputRef.current) {
+          inputRef.current.focus();
+        }
+      });
+  }
+
+  function deleteTodo(todoId: number) {
+    setLoadingTodoId(todoId);
+    deleteTodos(todoId)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        );
+        setCounterOfActiveTodos(currentCount => currentCount - 1);
+      })
+      .catch(err => setError('Unable to delete a todo' + err.message))
+      .finally(() => setLoadingTodoId(null));
+  }
+
+  function handleDeleteAllCompleted() {
+    const allCompletedTodos = todos.filter(todo => todo.completed);
+
+    allCompletedTodos.map(todo => deleteTodo(todo.id));
+  }
+
+  function updateTodoStatus(todoId: number, completed: boolean) {
+    const todoToUpdate = todos.find(todo => todo.id === todoId);
+
+    if (!todoToUpdate) {
+      return;
+    }
+
+    setLoadingTodoId(todoId);
+    updateTodos({ ...todoToUpdate, completed })
+      .then(updatedTodo => {
+        setTodos(currentTodos =>
+          currentTodos.map(todo =>
+            todo.id === updatedTodo.id ? updatedTodo : todo,
+          ),
+        );
+      })
+      .catch(err => setError('Unable to delete a todo' + err.message))
+      .finally(() => setLoadingTodoId(null));
+  }
+
+  const handleFilterChange = (newFilter: Filter) => setFilter(newFilter);
+
+  const filteredTodos = todos.filter(todo => {
+    switch (filter) {
+      case Filter.ACTIVE:
+        return !todo.completed;
+      case Filter.COMPLETED:
+        return todo.completed;
+      default:
+        return true;
+    }
+  });
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          todos={todos}
+          loading={loading}
+          inputRef={inputRef}
+          newTodoTitle={newTodoTitle}
+          setNewTodoTitle={setNewTodoTitle}
+          addTodo={addTodo}
+        />
+        <TodoList
+          filteredTodos={filteredTodos}
+          updateTodoStatus={updateTodoStatus}
+          deleteTodo={deleteTodo}
+          loadingTodoId={loadingTodoId}
+        />
+        {todos.length > 0 && (
+          <Footer
+            counterOfActiveTodos={counterOfActiveTodos}
+            filter={filter}
+            todos={todos}
+            handleFilterChange={handleFilterChange}
+            handleDeleteAllCompleted={handleDeleteAllCompleted}
+          />
```

Comment:
Do we want not to render only the footer if we don't have any todos?

Reply:
yes? why not

Reply:
Why do we need to render a todo list if there are no todos?

Reply:
aaaaa